### ISSUE TITLE
Use better column names in join table for implicit m-n relations

### ISSUE THREAD

**Issue by livthomas** [prisma/prisma#10243]
## Problem

I have the following schema which includes an implicit many-to-many relation:

```prisma
model Permission {
  id    Int    @id @default(autoincrement())
  name  String
  users User[]
}


model User {
  id             Int          @id @default(autoincrement())
  userName       String
  hashedPassword String
  permissions    Permission[]

  @@index([userName])
}

```

Prisma creates a join table with columns called `A` and `B`. When I inspect this database in some database tool, it is not clear which column represents which table in this relation.

## Suggested solution

Prisma should use a combination of a table name and an ID column name when naming columns of the join table. In this case, these columns would be called `permission_id` and `user_id`.

## Alternatives

Or Prisma could allow us to specify column names in `@relation` attribute:

```prisma
model Permission {
  id    Int    @id @default(autoincrement())
  name  String
  users User[] @relation(map: ["permission_id"])
}

model User {
  id             Int          @id @default(autoincrement())
  userName       String
  hashedPassword String
  permissions    Permission[] @relation(map: ["user_id"])

  @@index([userName])
}

```

## Additional context

I use Prisma 3.4.1 with SQLite database.

Related issues:
* https://github.com/prisma/prisma/issues/3381
* https://github.com/prisma/prisma/issues/5234
* https://github.com/prisma/prisma/issues/5369

**Comment by acisser** [comment]
Please take a look at this feature request since this is basic Object Relational functionality. The default column name convention Prisma is proposing for these columns is not standard and should be customizable to support any other convention.  Declaring models for these middle tables is overkilling and impacts not only all related query DSL but also domain logic code when retrieving values from objects returned by queries.

It's not the end of the world if it's not implemented but this as well as a few other limitations observed during only 3 weeks of exploration is making me think that we are still 1 or 2 years away from considering this tool as an all-terrain ORM

Hoping this will not be so hard to achieve. 

Congratulations to all of you that have put some effort into building this toolset that looks promising.

**Comment by jjosef** [comment]
I second this approach. Prisma does not always take the most efficient approach at complex queries, so `queryRaw` comes in handy pretty frequently. Being forced to use this `A` and `B` column structure is confusing when writing raw queries. At a bare minimum could we at least name the columns the same as the Model's mapped name?

**Comment by acisser** [comment]
The decision of what side of the relationship is A/B should also be configurable at least with the current approach. 
Right now the decision is driven by the alphabetical order of the model names which is not so deductive.

**Comment by brice-morin** [comment]
I agree with @jjosef suggestion to name the column `Permission` and `User` (referring to the example provided in this issue), rather than A and B. The A and B convention is not very refactoring friendly/future proof. For example, in our prisma model, we have one specific N-M association table that is implicit, but we fear that one day we will need to add an additional column for that association table, which would force us to make the implicit table implicit in the model, and obviously, we do not want columns to be called A and B, but rather called `Permission` and `User` (or similar). One way or another, if we need to do so, we will manage (make a new table rather than using the current one and do a data migration, use the new table instead, and finally drop the implicit table), but it would have been nicer to be able to switch from implicit to explicit association tables (nearly) seamless...

IMHO, generated code should be  as close as possible to code written by an expert. Of course, code generators need to be systematic, etc, cannot cope with each and every opinions and conventions experts may have and use. And @jjosef suggestions seems to be a nice step in that direction. I would suspect it would not incur any significant overhead for any of the DB supported by Prisma. Also, code generators should ideally not come with a vendor lock-in i.e., if one day one wants to go SQL-only rather than using Prisma (which I would not recommend😺), the SQL code needs to be as explicit as possible, not arbitrarily obscure for no good reasons.

**Comment by hassan-naveed-dev** [comment]
Mapping A & B column names are confusing when you are supposed to raw query. Prisma should have the facility to rename these columns

**Comment by hassanhnaveed** [comment]
Mapping A & B column names are confusing when you are supposed to raw query. Prisma should have the facility to rename these columns

**Comment by belizar** [comment]
I stumbled upon this today, please consider prioritize this feature.

**Comment by rmartins0** [comment]
Bump. I also came across this issue. I'm surprised Prisma uses obscure names. 
Does anyone know if we can specify explict join table models and colum names?

**Comment by JVKdouk** [comment]
Bump, we really need a way to override the column name in implicit M-to-N tables.

**Comment by deco190igc** [comment]
up

**Comment by datoslabs** [comment]
Bump.  I use Prisma to manage schema and ORM for front end; however, parts of the backend data pipeline uses N8N that relies on raw SQL for CRUD, it is very confusing and fragile with the current generic column naming convention "A" and "B".

**Comment by jflygare** [comment]
+1. For many of the reasons mentioned above.
